realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +342,41 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread util transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} gets called.
+     *
+     * @return {@code true} if transactions are committed to this Realm, {@code false} if
+     * {@link #stopWaitForChange()} gets called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        sharedGroupManager.setWaitForChangeEnabled(true);
+        // add this to wait list for batch stop
+        addRealmToWaitList(this);
+        boolean hasChanged = sharedGroupManager.waitForChange();
+        // remove this from wait list
+        removeRealmFromWaitList(this);
+        if (hasChanged) {
+            sharedGroupManager.advanceRead();
```

Comment:
Not to my knowledge. Only use case seems to be unit tests. Which was also at least me and @nhachicha agreed that just removing it altogether for `waitForChange` would be sensible (since waitForChange as better semantics for non-looper threads).